lib/list.gi

```diff
@@ -3916,10 +3916,13 @@ InstallGlobalFunction(PlainListCopy, function( list )
     tnum := TNUM_OBJ(copy);
     if FIRST_LIST_TNUM > tnum or LAST_LIST_TNUM < tnum then
         copy := PlainListCopyOp( copy );
+    elif FIRST_PLIST_TNUM > tnum or LAST_PLIST_TNUM < tnum then
+        copy := PLAIN_LIST_COPY( copy );
```

Comment:
I think there's more wrong with `PlainListCopy`, and I am not sure what the original intention for this function were -- it's possible that the author meant "plain list" as "native GAP list", which would explain the current code. (The original authors turns out to be @stevelinton who added it 1998-08-28 in the "old repository commit b6f3672c8a5b59f7248c8b3e8d5f2402ff4bca9c").

But, OK, I agree we probably want "real" plain lists (but let's inspect existing callers first, to make sure?)

Also, looking closer at `PlainListCopyOp` methods there are, several of those seem to be *wrong*. E.g.
```gap
InstallMethod( PlainListCopyOp, "for a GF2 vector",
        true, [IsGF2VectorRep and IsSmallList ],
        0, function( v )
    PLAIN_GF2VEC(v);
    return v;
end);
```
This modifies `v` in place!

Thinking some more about it, if we truly want a generic way to copy any list to a plain list, the "obvious" one to do it is:  `List(somelist, x -> x);` but of course that may have some special methods for "optimizing" this, so better is:
```gap
  n := Length(oldlist);
  newlist := EmptyPlist(n);
  for i in [1..n] do newlist[i] := oldlist[i]; end;
```
and then we can install better methods if we really think it's worth it (but beware: using method dispatch here can be a major performance killer...)

Reply:
You are right, I definitely didn't pay this bit enough attention. I think calling PLAIN_LIST_COPY in the kernel should handle the "efficient" versions, then just doing something boring.

I think what happened is we had PLAIN_LIST in the kernel, added PLAIN_LIST_COPY as a variant of that, and never connected that to the (undocumented) GAP-level PlainListCopy.

Reply:
I'm going to pull this bit out in a separate PR.